eng/scripts/Language-Settings.ps1

```diff
@@ -0,0 +1,42 @@
+$Language = "go"
+$LanguageDisplayName = "go"
+
+function Get-AllPackageInfoFromRepo ($serviceDirectory)
+{
+    $allPackageProps = @()
+    $searchPath = Join-Path $RepoRoot "sdk"
+    if ($serviceDirectory)
+    {
+        $searchPath = Join-Path $searchPath $serviceDirectory 
+    }
+
+    $pkgFiles = Get-ChildItem -Path $searchPath -Include "go.mod" -Recurse
+
+    foreach ($pkgFile in $pkgFiles)
+    {
+        $pkgPath = $pkgFile.DirectoryName
+        $pkgName = $pkgFile.Directory.Name
+        $pkgVersion = ""
+        $serviceDirectory = $pkgName
+
+        if ($pkgPath -match ".*\\sdk\\(?<serviceDir>.*)")
+        {
+            $serviceDirectory = $Matches["serviceDir"]
+        }
+        $versionFile = Join-Path $pkgPath "version.go"
+
+        if (Test-Path $versionFile)
+        {
+            $versionFileContent = (Get-Content -Path $versionFile -Raw) -replace '"', '\"'
```

Comment:
Do we need to directly read the go version file? Can we not have that be picked up by the go program? 

If for whatever reason we don't need to read any other properties beyond version we should simply use a regex here instead of invoking go at all.

Reply:
I also recommend we just do a regex matching here instead of relying on a Go program to get the constant. Currently our repo of azure-sdk-for-go is mixing the module and non-module programs, you could encounter something wrong when you `go run` something.